options/options.go

```diff
@@ -14,23 +15,35 @@ type Options struct {
 	DryRun    bool
 	ParseTime bool
 	Config    *string
+	Edit      *string
 }
 
 var OptionsToPrint = `
   -h, --help              Prints this help message
-  -c, --config            Reads the config from custom path
+  -c, --config [path]     Reads the config from custom path
   -d, --dry-run           Prints bindings and their commands and exits
   -k, --kill	          Gracefully kills every running instances of dxhd
   -p, --parse-time        Prints how much time parsing a config took
   -r, --reload	          Reloads every running instances of dxhd
-  -v, --version	          Prints current version of program`
+  -v, --version	          Prints current version of program
+  -e, --edit [file]	      Shortcut to edit a file in dxhd's config folder. Opens dxhd.sh if file is empty.`
 
 func Parse() (opts Options, err error) {
 	osArgs := os.Args[1:]
 
 	skip := false
 
-toplevel:
+	notEnoughArgsErr := errors.New("Not enough arguments given")
+	readNextArg := func(index int, optional bool) (*string, error) {
+		if index == len(osArgs)-1 || strings.HasPrefix(osArgs[index+1], "-") {
+			if optional {
+				return nil, nil
+			}
+			return nil, notEnoughArgsErr
+		}
+		return &osArgs[index+1], nil
+	}
```

Comment:
Follow the style of other dxhd functions:
```suggestion
	readNextArg := func(index int, optional bool) (arg *string, e error) {
		if (index == len(osArgs)-1 || strings.HasPrefix(osArgs[index+1], "-")) && !optional {
			e = notEnoughArgsErr
		} else {
			arg = &osArgs[index+1]
		}
		return
	}
```

Reply:
Or, even better:
```suggestion
	readNextArg := func(index int, optional bool) (arg *string, e *error) {
		if (index == len(osArgs)-1 || strings.HasPrefix(osArgs[index+1], "-")) && !optional {
			e = new(error)
		} else {
			arg = &osArgs[index+1]
		}
		return
	}
``` so you can also get rid of `notEnoughArgsErr` variable and replace `if err == notEnoughArgsErr` with `if err != nil`.

Reply:
Instead of `new(error)`, it should be `errors.New("not enough arguments passed to flags")` though.